src/designer/elsa-workflows-studio/src/components/controls/elsa-input-tags/elsa-input-tags-dropdown.tsx

```diff
@@ -0,0 +1,112 @@
+import { Component, h, Prop, Event, EventEmitter, State, Watch } from '@stencil/core';
+import { MultiTextDefinition } from '../../../models/domain';
+
+@Component({
+    tag: 'elsa-input-tags-dropdown',
+    styleUrl: 'elsa-input-tags-dropdown.css',
+    shadow: false,
+})
+export class ElsaInputTagsDropdown {
+
+    @Prop() fieldName?: string;
+    @Prop() fieldId?: string;
+    @Prop() placeHolder?: string = 'Add tag';
+    @Prop() values?: Array<string | MultiTextDefinition> = [];
+    @Prop() dropdownValues?: Array<MultiTextDefinition> = [];
+    @Event({ bubbles: true }) valueChanged: EventEmitter<Array<string | MultiTextDefinition>>;
+    @State() currentValues?: Array<MultiTextDefinition> = [];
+    @State() dropdownTags?: Array<MultiTextDefinition> = [];
+
+    @Watch('values')
+    valuesChangedHandler(newValue: Array<string | MultiTextDefinition>) {
+        let values: Array<MultiTextDefinition> = [];
+
+        newValue.forEach(value => {
+            this.dropdownValues.forEach(tag => {
+                if (value === tag.value) {
+                    values.push(tag);
+                }
+            })
+        })
+        this.currentValues = values || [];
```

Comment:
How about we change this block to this:

```javascript
this.currentValues = newValue.filter(value => this.dropdownValues.findIndex(tag => tag.value == value) > -1);
```

The same principle can be applied elsewhere of course.

Reply:
Good suggestion to use builtin methods if possible.
However the fact that `newValues` is `Array<string | MultiTextDefinition>` and `currentValues` is `Array<MultiTextDefinition>` makes it error on compile with your suggestion.

Will think over night to see if there is better way to type these members and implement this better.